packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -66,7 +66,6 @@ export class B3Propagator implements HttpTextPropagator {
     const traceIdHeader = getter(carrier, X_B3_TRACE_ID);
     const spanIdHeader = getter(carrier, X_B3_SPAN_ID);
     const sampledHeader = getter(carrier, X_B3_SAMPLED);
-    if (!traceIdHeader || !spanIdHeader) return context;
     const traceId = Array.isArray(traceIdHeader)
```

Comment:
The root of this issue was that `Array.isArray` turns our `unknown` into an `any` which made the type system not complain like we would have expected.